src/main/java/com/twilio/http/NetworkHttpClient.java

```diff
@@ -52,6 +52,22 @@ public NetworkHttpClient() {
     }
 
     /**
+     * Create a new HTTP Client using custom configuration
+     */
+    public NetworkHttpClient(HttpClientBuilder clientBuilder){
+        Collection<Header> headers = Lists.<Header>newArrayList(
+                new BasicHeader("X-Twilio-Client", "java-" + Twilio.VERSION),
+                new BasicHeader(HttpHeaders.USER_AGENT, "twilio-java/" + Twilio.VERSION + " (" + Twilio.JAVA_VERSION + ") custom"),
+                new BasicHeader(HttpHeaders.ACCEPT, "application/json"),
+                new BasicHeader(HttpHeaders.ACCEPT_ENCODING, "utf-8")
+        );
+
+        client = clientBuilder
+                .setDefaultHeaders(headers)
+                .build();
+    }
+
+    /**
```

Comment:
Hi @efossier , great stuff! But I'm afraid TwilioRestClient will still try to create NetworkHttpClient with default constructor anyway. See https://github.com/twilio/twilio-java/blob/31279b95d2dd38ede6ca58edc5c8c7664e0bb3f5/src/main/java/com/twilio/http/TwilioRestClient.java#L66
And on Google App Engine it will fail right then when trying to do network related stuff in constructor.